api/app/hfi/hfi_calc.py

```diff
@@ -202,18 +206,84 @@ async def calculate_latest_hfi_results(request: HFIResultRequest):
         async for station_daily in dailies_generator:
             dailies.append(station_daily)
 
-        prep_delta = valid_date_range.end_date - valid_date_range.start_date
-        prep_days = prep_delta.days + 1  # num prep days is inclusive
-
-        results = calculate_hfi_results(request.planning_area_fire_starts,
-                                        dailies, prep_days,
+        results = calculate_hfi_results(fire_centre_fire_start_ranges,
+                                        request.planning_area_fire_starts,
+                                        fire_start_lookup,
+                                        dailies, valid_date_range.days_in_range(),
                                         request.selected_station_code_ids,
                                         area_station_map,
                                         valid_date_range.start_date)
         return results, valid_date_range
 
 
-def calculate_hfi_results(planning_area_fire_starts: Mapping[int, FireStartRange],  # pylint: disable=too-many-locals
+def build_fire_start_prep_level_lookup(orm_session) -> Dict[int, Dict[int, int]]:
```

Comment:
What's the reason behind introducing the `Dict` type here instead of the existing `Mapping` type?

Reply:
I am SOOOOOOO excited you asked. Short answer: because it's returning a dict, not a map. If it were Mapping, it would inherit from collection, but {} or dict() doesn't.

If you peak at the definitions in: /usr/lib/python3.8/typing.py, you'll see
```
Dict = _alias(dict, (KT, VT), inst=False)
Mapping = _alias(collections.abc.Mapping, (KT, VT_co))
```

Reply:
Cool thanks. Should it return a map though? Isn't a `Mapping` an immutable collection? That's what this should be when constructed, yeah?

Reply:
Oh hell yeah! Super valid comment. Yes! It should totally be that! I'll make that change - and then I'd like to merge this in.

Reply:
We'd have to go to a bunch of effort to make an immutable object to represent the prep level lookup. It's a fair point that the intention is that the result be immutable - but that's a lot of work.